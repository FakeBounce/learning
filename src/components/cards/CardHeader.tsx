import { Box, Typography } from '@mui/material';
import { memo, ReactNode } from 'react';
import ActionButton from '@src/components/lms/ActionButton';

interface CardHeaderActionProps {
  action: () => void | VoidFunction;
  actionType?: 'cancel' | 'update' | 'action';
  actionText: ReactNode;
}

interface CardHeaderProps {
  headerText: ReactNode;
  headerColor?: string;
  actions?: CardHeaderActionProps[] | null;
}

function CardHeader({ headerText, headerColor, actions = null }: CardHeaderProps) {
  return (
    <Box
      display="flex"
      gap={2}
      flexDirection={[actions && actions.length > 1 ? 'column' : 'row', 'row']}
    >
      <Typography
        sx={{
          fontSize: (theme) => theme.typography.h3.fontSize,
          fontWeight: (theme) => theme.typography.fontWeightBold,
          color: (theme) => headerColor ?? theme.palette.grey[800]
        }}
      >
        {headerText}
      </Typography>
      {actions && (
        <Box display="flex" gap={2}>
          {actions.map((action, index) => (
            <ActionButton
              key={index}
              actionType={action.actionType ?? 'action'}
              sx={{ textTransform: 'none' }}
              onClick={action.action}
            >
              {action.actionText}
            </ActionButton>
          ))}
        </Box>
      )}
    </Box>
  );
}

export default memo(CardHeader);
