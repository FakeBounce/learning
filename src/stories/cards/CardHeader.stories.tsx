import { Meta, StoryObj } from '@storybook/react';
import CardHeader from '@src/components/cards/CardHeader';
import { fn } from '@storybook/test';
import palette from '@src/theme/palette';

const meta: Meta = {
  title: 'Example/CardHeader',
  component: CardHeader,
  tags: ['autodocs'],
  parameters: {
    layout: 'centered'
  }
};

export default meta;
type Story = StoryObj<typeof CardHeader>;

const CardHeaderTemplate: Story = {
  render: (args: any) => {
    return <CardHeader {...args} />;
  }
};

export const Default = {
  ...CardHeaderTemplate,
  args: {
    headerText: 'Header'
  }
};

export const ColoredHeader = {
  ...CardHeaderTemplate,
  args: {
    headerText: 'Colored header',
    headerColor: palette().secondary.main
  }
};

export const WithSingleAction = {
  ...CardHeaderTemplate,
  args: {
    headerText: 'Header with single action',
    actions: [
      {
        actionText: 'Action',
        action: fn(() => console.log('Action clicked'))
      }
    ]
  }
};

export const WithMultipleAction = {
  ...CardHeaderTemplate,
  args: {
    headerText: 'Header with multiple action',
    actions: [
      {
        actionText: 'Action 1',
        action: fn(() => console.log('Action 1 clicked'))
      },
      {
        actionText: 'Action 2',
        actionType: 'update',
        action: fn(() => console.log('Action 2 clicked'))
      }
    ]
  }
};
