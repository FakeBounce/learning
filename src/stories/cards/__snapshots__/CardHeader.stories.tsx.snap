// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example/CardHeader ColoredHeader smoke-test 1`] = `
<div class="MuiBox-root css-10egq61">
  <p class="MuiTypography-root MuiTypography-body1 css-thswzk-MuiTypography-root">
    Colored header
  </p>
</div>
`;

exports[`Example/CardHeader Default smoke-test 1`] = `
<div class="MuiBox-root css-10egq61">
  <p class="MuiTypography-root MuiTypography-body1 css-tjb74-MuiTypography-root">
    Header
  </p>
</div>
`;

exports[`Example/CardHeader WithMultipleAction smoke-test 1`] = `
<div class="MuiBox-root css-10egq61">
  <p class="MuiTypography-root MuiTypography-body1 css-tjb74-MuiTypography-root">
    Header with multiple action
  </p>
  <button class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-121cmne-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
          tabindex="0"
          type="button"
          id=":r1:"
  >
    Action 1
    <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
    </span>
  </button>
  <button class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1gn5gyf-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
          tabindex="0"
          type="button"
          id=":r2:"
  >
    Action 2
    <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
    </span>
  </button>
</div>
`;

exports[`Example/CardHeader WithSingleAction smoke-test 1`] = `
<div class="MuiBox-root css-10egq61">
  <p class="MuiTypography-root MuiTypography-body1 css-tjb74-MuiTypography-root">
    Header with single action
  </p>
  <button class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-121cmne-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
          tabindex="0"
          type="button"
          id=":r0:"
  >
    Action
    <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
    </span>
  </button>
</div>
`;
